src/events/types/inventory/PrepareItemCraftEvent.ts

```diff
@@ -0,0 +1,30 @@
+import { InventoryEvent } from "./InventoryEvent";
+import { Inventory } from "../../../inventory/Inventory";
+import { Recipe } from "../../../inventory/Recipe";
+import { CraftingInventory } from "../../../inventory/types";
+
+export class PrepareItemCraftEvent extends InventoryEvent {
+
+  public static getBukkitClasspath(): string {
+    return 'org.bukkit.event.inventory.PrepareItemCraftEvent';
+  }
+
+  getInventory(): CraftingInventory {
+    return Inventory.fromJava<CraftingInventory>(this.toJava().getInventory());
+  }
+
+  /**
+   * Get the recipe that has been formed. If this event was triggered by a tool repair, this will be a temporary shapeless recipe representing the repair.
+   */
+  getRecipe(): Recipe {
+    const recipe = this.toJava().getRecipe();
+    return recipe && Recipe.fromJava(recipe);
+  }
```

Comment:
If the recipe value can be null, update the return type to `Recipe | null` probably